1 Create basic pages for users' creation

RomuloDevelop opened the pull request:
a. Login with username
  b. Register with name, surname and username
  c. Fields' validations for max and min length
2 Creation of Post pages
  a. Special form with post data
  b. Post item component and list for all posts available
  c. Add Reverse Geocoding with Radar API
  d. Add logic for likes
3 Creation of profile page
  a. User data form with avatar selection
  b. The same fields as in Register's page
4 Add redux store with all the data models of posts and users
  a. Add special selectors for merge of user and post models and search filter
  b. Add persistant storage except for search filter

RomuloDevelop committed:
1 Create basic pages for users' creation
  a. Login with username
  b. Register with name, surname and username
  c. Fields' validations for max and min length
2 Creation of Post pages
  a. Special form with post data
  b. Post item component and list for all posts available
  c. Add Reverse Geocoding with Radar API
  d. Add logic for likes
3 Creation of profile page
  a. User data form with avatar selection
  b. The same fields as in Register's page
4 Add redux store with all the data models of posts and users
  a. Add special selectors for merge of user and post models and search filter
  b. Add persistant storage except for search filter

RomuloDevelop committed:
1 Add optimizations with useCallback
2 Fix ImagePicker component error when selecting same image
3 Add first render hook
4 Add netlify url to REDME

RomuloDevelop committed:
Remove no used vars

RomuloDevelop committed:
Add .redirect file for netlify

RomuloDevelop committed:
Add option for generation of batch

RomuloDevelop committed:
Addin load batch logic

RomuloDevelop committed:
Create UserTag

RomuloDevelop committed:
Add posts count selector for actual user

christhoval06 commented:
Hola @RomuloDevelop podrías dejar todo en `dev` y crear un PR a `main` para que sea más sencillo el tema del review.

christhoval06 commented on package.json:
```diff
@@ -3,18 +3,29 @@
   "version": "0.1.0",
   "private": true,
   "dependencies": {
+    "@headlessui/react": "^1.5.0",
```
Recuerda que debemos separar las `devDependencies`, es to nos ayuda a organizar nuestros pipelines, scripts y sobre todo nuestro code

christhoval06 commented on src/guards/AuthGuard.tsx:
```diff
@@ -0,0 +1,18 @@
+  const navigate = useNavigate()
+  const location = useLocation()
+  useEffect(() => {
+      if (location.pathname !== '/login' && location.pathname !== '/register' && !actualUser) {
```
```
// ... en rutas.config.ts
export const LOGIN = '/login'
export const REGISTER = '/register'
export const rutasPublicas = [LOGIN, REGISTER]

// ... en AuthGuard.tsx
if(!actualUser && !rutasPublicas.includes(location.pathname)){
...
}

```

christhoval06 commented on src/App.tsx:
```diff
@@ -1,25 +1,23 @@
-        </a>
-      </header>
-    </div>
+    <AuthGuard>
```
Para evitar el uso de un componente wrapper para validar la auth. Podrías utilizar un `PrivateRoute` que se encarga de solo esa tarea.

christhoval06 commented on src/components/AvatarSelector/index.tsx:
```diff
@@ -0,0 +1,45 @@
+import useFirstRender from '../../hooks/useFirstRender'
+import ImagePicker, { HandleImage } from '../ImagePicker'
+import './AvatarSelector.scss'
+
```
definición de `Props`

```
interface Props {...}
// o
type Props = {...}
```

christhoval06 commented on src/pages/Home/index.tsx:
```diff
@@ -0,0 +1,49 @@
+import UserTag from '../../components/UserTag';
+
+const Home = ({posts}: {posts: PostUser[]}) => {
+  let search = useSelector(selectSearch)
```
preferiblemente `const`

christhoval06 commented on src/components/PostItem/index.tsx:
```diff
@@ -0,0 +1,108 @@
+import Card from '../Card'
+import './PostItem.scss'
+
+const PostItem = ({...props}: PostUser) => {
```
`...` no es necesario

christhoval06 commented on src/components/PostItem/index.tsx:
```diff
@@ -0,0 +1,108 @@
+}
+
+const PostDescription = ({message}: {message: string}) => {
+  let messageBox = useRef<HTMLParagraphElement>(null)
```
mejor `const`

christhoval06 replied:
Tienes algún problema con los `const`?

christhoval06 commented on src/components/PostItem/index.tsx:
```diff
@@ -0,0 +1,108 @@
+
+const PostItem = ({...props}: PostUser) => {
+  const actualUser = useSelector(selectUser) as User
+  const time = moment(props.create_at).from(new Date())
```
`dayjs` es mas eficiente

christhoval06 commented on src/components/PostForm/index.tsx:
```diff
@@ -0,0 +1,94 @@
+
+
+const PostForm = () =>  {
+  let author = useSelector(selectUser) as User
```
los `let`s

christhoval06 commented on src/components/MainHeader/index.tsx:
```diff
@@ -0,0 +1,195 @@
+
+type SearchEvent = FormEvent<HTMLInputElement>
+
+const Header = () => {
```
intentemos componetizar y tendremos archivos mas manejables.

christhoval06 reviewed (commented):
Muy buen trabajo, solo detalles.

RomuloDevelop committed:
Add visual confirmation of batch's update

RomuloDevelop committed:
Fixin header index for webkit

RomuloDevelop committed:
1 Remove let variables
2 Add type Prop notation for components
3 Add router animations

RomuloDevelop committed:
Remove unused variables

RomuloDevelop committed:
1 Drop moment implementation for dayjs
2 Create new component for modal search
3 Create error handling for batch creation
